### PR TITLE
add global scrape interval config option

### LINKED ISSUE #99: add global_scrape_interval as a config option
### Enhancement Proposal

We should add a Juju config option to set the `global_scrape_interval`, in the same way as we do for the scrape timeout.

### PR DESCRIPTION
**Pull request by lucabello**
## Issue
Closes #99.

### CODE REVIEW

File: src/grafana_agent.py [hunk 1]
```diff
@@ -615,6 +615,7 @@ def _generate_config(self) -> Dict[str, Any]:
                 "wal_directory": "/tmp/agent/data",
                 "global": {
                     "scrape_timeout": self.model.config.get("global_scrape_timeout"),
+                    "scrape_interval": self.model.config.get("scrape_interval"),
```
Comment (mmkay) [inline src/grafana_agent.py:618]:
```suggestion
                    "scrape_interval": self.model.config.get("scrape_interval"),
                    "scrape_interval": self.model.config.get("global_scrape_interval")
```

**Review by mmkay — CHANGES_REQUESTED** [review]
One naming mismatch, other than that looks good :)

**Review by PietroPasotti — DISMISSED** [review]
required: document the input format
optional: validate the input

File: src/grafana_agent.py (hunk 1 above)
Comment (lucabello) [reply src/grafana_agent.py:618]: `yaml.dump()` would conver the `None` to a `null`, making the configuration invalid I'd assume

**Review by mmkay — APPROVED** [review]

File: config.yaml [hunk 2]
```diff
@@ -14,3 +14,7 @@ options:
     description: The global metrics scrape timeout.
     type: string
     default: "10s"
+  global_scrape_interval:
+    description: The global metrics scrape interval.
```
[the remark this answers is not in the document]
Comment (lucabello) [reply config.yaml]: I honestly think that if you see two default values being `10s` and `1m`, and you pass `42 days and 2 seconds` and expect it to work, you're probably gonna struggle to set anything up; not even the [grafana-agent docs](https://grafana.com/docs/agent/latest/static/configuration/metrics-config/#global_config) mentions that

I have added a list of supported units however :)

**Comment by lucabello** [comment]
A typo breaks the configuration, just like any other config option we have; there's nothing we can do about that except asking Juju for config option validation.

This has the same problem that the discussion for Mimir HA has, where if we try to correct a typo we open up numerous problems:
* do we try to make an educated guess on what the user meant? too risky and unpredictable;
* do we not apply the config and keep the previous value? we'd need to store the config values, or they'd be lost in an upgrade

Unfortunately I believe there's no good solution except an upstream one :)

**Pull request merged**